src/dune_engine/scheduler.ml

```diff
@@ -993,31 +971,39 @@ end = struct
     | Ok -> res
 end
 
-let go config f =
-  let t = prepare config ~polling:false in
-  let res =
-    let f = Rpc0.with_rpc_serve t.rpc f in
-    Run_once.run_and_cleanup t f
-  in
-  match res with
-  | Error (Exn exn) -> Exn_with_backtrace.reraise exn
-  | Ok res -> res
-  | Error (Got_signal | Never) -> raise Dune_util.Report_error.Already_reported
-
-let maybe_clear_screen (config : Config.t) =
-  match config.terminal_persistence with
-  | Clear_on_rebuild -> Console.reset ()
-  | Preserve ->
-    Console.print_user_message
-      (User_message.make
-         [ Pp.nop
-         ; Pp.tag User_message.Style.Success
-             (Pp.verbatim "********** NEW BUILD **********")
-         ; Pp.nop
-         ])
-
-let poll (config : Config.t) ~once ~finally =
-  let t = prepare config ~polling:true in
+module Build = struct
```

Comment:
I feel like this module is strangely named and has strangely named members, especially `Build.to_handler` makes no sense to me. Anyway, I don't have a clear enough idea of where we want to end up, so I won't try to propose names here. If you can improve anything, please do.